crates/rytest/src/phases/execution.rs

```diff
@@ -1,12 +1,102 @@
 use anyhow::Result;
 use pyo3::prelude::*;
-use pyo3::types::{PyIterator, PyMapping, PyString};
+use pyo3::types::{PyIterator, PyMapping};
 use pyo3::{indoc::indoc, types::PyTuple};
+use std::path::PathBuf;
 use std::{env, fs, path::Path, sync::mpsc};
 
 use crate::python;
 use crate::TestCase;
 
+fn file_contains_fixture(file_path: &Path, fixture_name: &str) -> bool {
+    if let Ok(content) = fs::read_to_string(file_path) {
+        let def_pattern = format!("def {}(", fixture_name);
+        let fixture_pattern = format!("fixture(name=\"{}\")", fixture_name);
+        return content.contains(&def_pattern) || content.contains(&fixture_pattern);
+    }
+    false
+}
+
+fn find_fixture_reversely(
+    start_path: &Path,
+    fixture_name: &str,
+    stop_path: &PathBuf,
+) -> Option<PathBuf> {
+    let mut current_path = start_path.to_path_buf();
+
+    // Check if start_path is a file and search it first
+    if current_path.is_file() {
+        if file_contains_fixture(&current_path, fixture_name) {
+            return Some(current_path);
+        }
+        // Move to the parent directory if the start path was a file
+        current_path.pop();
+    }
+
+    // Begin directory traversal
+    loop {
+        //n!("Checking directory for fixture {}: {:?}", fixture_name, current_path);
```

Comment:
probably worth purging, or we can maybe make a verbosity level (I forget if we have that) and log this sort of stuff on high verbosity levels?